main.tf

```
@@ -35,7 +30,6 @@ resource "aws_customer_gateway" "this" {
 }
 
 resource "aws_vpn_connection" "this" {
-  count = var.create_vpn_connection ? length(var.cgw_ip_addresses) : 0
 
   vpn_gateway_id      = aws_vpn_gateway.this[0].id
```

We are removing `count` entirely from `aws_vpn_gateway`, so the resource is no longer a set and we need to stop indexing into it...
```suggestion
  vpn_gateway_id      = aws_vpn_gateway.this.id
```